### PR TITLE
Add MH docstring

### PR DESCRIPTION
**Pull request by cpfiffer**
Adds a docstring explaining all the cool fiddly bits of `MH`.

### CODE REVIEW

File: src/inference/mh.jl [hunk 1]
```diff
@@ -9,6 +9,145 @@ end
 proposal(p::AdvancedMH.Proposal) = p
 proposal(cov::AbstractMatrix) = AdvancedMH.RandomWalkProposal(MvNormal(cov))
 
+"""
+    MH(space...)
+
+Construct a Metropolis-Hastings sampler.
```
Comment (xukai92) [inline src/inference/mh.jl:15]: `sampler` -> `algorithm`?

File: src/inference/mh.jl [hunk 2]
```diff
@@ -9,6 +9,145 @@ end
...
+    MH(
+        :s => InverseGamma(2, 3),
+        :m => x -> Normal(x, 1)
+    ), 
+    1_000
+)
+mean(chain)
+````
+
+Providing a variance covariance matrix will cause `MH` to perform random-walk
```
Comment (xukai92) [inline src/inference/mh.jl:126]: `variance covariance`?

File: src/inference/mh.jl (hunk 1 above)
Comment (cpfiffer) [inline src/inference/mh.jl:15]:
```suggestion
Construct a Metropolis-Hastings algorithm. 
```

File: src/inference/mh.jl [hunk 3]
```diff
@@ -9,6 +9,145 @@ end
...
+sampling in the transformed space with proposals drawn from a multivariate
+normal distribution:
```
Comment (cpfiffer) [inline src/inference/mh.jl:128]:
```suggestion
Providing a covariance matrix will cause `MH` to perform random-walk
sampling in the transformed space with proposals drawn from a multivariate
normal distribution. The provided matrix must be positive semi-definite and square. Usage:
```

**Pull request merged**